azext_iot/central/commands_export.py

```diff
@@ -0,0 +1,136 @@
+from azext_iot.constants import CENTRAL_ENDPOINT
+from azext_iot.central.providers import CentralExportProvider
+from azext_iot.central.models.enum import ApiVersion
+from azext_iot.central.models.v1_1_preview import ExportV1_1_preview
+
+
+def get_dataExport_export(
+    cmd,
+    app_id: str,
+    export_id: str,
+    token=None,
+    central_dns_suffix=CENTRAL_ENDPOINT,
+    api_version=ApiVersion.v1_1_preview.value,
+) -> Union[dict, ExportV1_1_preview]:
+    provider = CentralExportProvider(
+        cmd=cmd, app_id=app_id, api_version=api_version, token=token
+    )
+
+    return provider.get_dataExport_export(
+        export_id=export_id, central_dnx_suffix=central_dns_suffix
+    )
+
+
+def delete_dataExport_export(
+    cmd,
+    app_id: str,
+    export_id: str,
+    token=None,
+    central_dns_suffix=CENTRAL_ENDPOINT,
+    api_version=ApiVersion.v1_1_preview.value,
+):
+    provider = CentralExportProvider(
+        cmd=cmd, app_id=app_id, api_version=api_version, token=token
+    )
+
+    provider.delete_dataExport_export(
+        export_id=export_id, central_dnx_suffix=central_dns_suffix
+    )
+
+
+def list_dataExport_exports(
+    cmd,
+    app_id: str,
+    token=None,
+    central_dns_suffix=CENTRAL_ENDPOINT,
+    api_version=ApiVersion.v1_1_preview.value,
+) -> List[Union[dict, ExportV1_1_preview]]:
+    provider = CentralExportProvider(
+        cmd=cmd, app_id=app_id, api_version=api_version, token=token
+    )
+
+    return provider.list_dataExport_exports(central_dns_suffix=central_dns_suffix)
+
+
+def add_dataExport_export(
+    cmd,
+    app_id: str,
+    export_id: str,
+    source,
+    destinations,
+    display_name,
+    enabled=True,
+    filter=None,
+    enrichments=None,
+    token=None,
+    central_dns_suffix=CENTRAL_ENDPOINT,
+    api_version=ApiVersion.v1_1_preview.value,
+) -> Union[dict, ExportV1_1_preview]:
+    export = {
+        "id": export_id,
+        "source": source,
+        "displayName": display_name,
+        "enabled": bool(enabled),
+    }
+
+    if filter is not None:
+        export.update({"filter": filter})
+
+    if enrichments is not None:
+        export.update(
+            {
+                "enrichments": utility.process_json_arg(
+                    content=enrichments, argument_name="enrichments"
+                )
+            }
+        )
+
+    if destinations is not None:
+        export.update(
+            {
+                "destinations": utility.process_json_arg(
+                    content=destinations, argument_name="destinations"
+                )
+            }
+        )
+
+    provider = CentralExportProvider(
+        cmd=cmd, app_id=app_id, api_version=api_version, token=token
+    )
+
+    return provider.add_dataExport_export(
+        export_id=export_id,
+        payload=export,
+        central_dnx_suffix=central_dns_suffix,
+    )
+
+
+def update_dataExport_export(
+    cmd,
+    app_id: str,
+    export_id: str,
+    content: str,
```

Comment:
Same weirdness to have update use content while create uses the params. Also would like to know limits of update

Reply:
I would rather didn't separate the content for create command, here is the partial content of export or destination payload, doesn't look like unified with create command right now, is there a rule that CLI teams defined that update must be per properties?